src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -467,6 +470,30 @@ public IEnumerable<Attachment> GetAttachmentsForDocument(DocumentsOperationConte
             }
         }
 
+        public IEnumerable<Attachment> GetAttachmentsForDocument(DocumentsOperationContext context, AttachmentType type, LazyStringValue lowerDocumentId)
+        {
+            var table = context.Transaction.InnerTransaction.OpenTable(AttachmentsSchema, AttachmentsMetadataSlice);
+            using (GetSliceFromLazyString(context, lowerDocumentId, out Slice lowerDocumentIdSlice))
+            using (GetAttachmentPrefix(context, lowerDocumentIdSlice, type, Slices.Empty, out Slice prefixSlice))
+            {
+                foreach (var sr in table.SeekByPrimaryKeyPrefix(prefixSlice, Slices.Empty, 0))
+                {
+                    var attachment = TableValueToAttachment(context, ref sr.Value.Reader);
+                    if (attachment == null)
+                        continue;
+
+                    attachment.Size = GetAttachmentStreamLength(context, attachment.Base64Hash);
+
+                    yield return attachment;
+                }
+            }
+        }
+
+        private static ByteStringContext.InternalScope GetSliceFromLazyString(DocumentsOperationContext context, LazyStringValue lowerDocumentId, out Slice lowerDocumentIdSlice)
```

Comment:
Not sure if this is worth a separate method.

Reply:
It isn't worth it, but this is required by the compiler, cannot use `yield` and `unsafe` together.

Reply:
Fixed this by adding a new `Slice.From` overload